runner/operator/alignment_pair_operator/v1_0_0/alignment_pair_operator.py

```diff
@@ -0,0 +1,458 @@
+        for sample in error_samples:
+            sample_name = sample.get("SM", "missingSampleName")
+            sample_id = sample.get("sample_id", "missingSampleId")
+            patient_id = sample.get("patient_id", "missingPatientId")
+            specimen_type = sample.get("specimen_type", "missingSpecimenType")
+            s.append("| " + sample_id + " | " + sample_name + " |" + patient_id + " |" + specimen_type + " |")
+            unformatted_s.append(sample_id + "\t" + sample_name + "\t" + patient_id + "\t" + specimen_type + "\n")
+
+        msg = """
+        Number of samples with error: {number_of_errors}
+
+        Error samples (also see error_sample_formatting.txt):
+        | IGO Sample ID | Sample Name / Error | Patient ID | Specimen Type |
+        {error_sample_names}
+        """
+
+        msg = msg.format(number_of_errors=str(len(error_samples)), error_sample_names="\n".join(s))
+
+        self.send_message(msg)
+
+        sample_errors_event = UploadAttachmentEvent(
+            self.job_group_notifier_id, "error_sample_formatting.txt", "".join(unformatted_s)
+        ).to_dict()
+        send_notification.delay(sample_errors_event)
+
+    def format_sample_name(self, *args, **kwargs):
+        return format_sample_name(*args, **kwargs)
+
+    def on_job_fail(self, run):
+        cmo_sample_name = run.tags.get("sampleNameTumor")
+        files = FileRepository.filter(queryset=self.files, metadata={"cmoSampleName": cmo_sample_name})
+        if files:
+            qc_report = files[0].metadata["qcReports"]
+            sample_id = files[0].metadata["sampleId"]
+            """
+            {
+                "comments": "Suboptimal quantity",
+                "qcReportType": "LIBRARY",
+                "IGORecommendation": "Try",
+                "investigatorDecision": "Continue processing"
+            }
+            """
+            report_str = ""
+            for report in qc_report:
+                report_str += "{comments}\t{qc_report_type}\t{igo_recommendation}\t{investigator_decision}\n".format(
+                    comments=report["comments"],
+                    qc_report_type=report["qcReportType"],
+                    igo_recommendation=report["IGORecommendation"],
+                    investigator_decision=report["investigatorDecision"],
+                )
+            msg = """
+cmoSampleId: {cmo_sample_name}
+sampleId: {sample_id}
+Comments\tQC Report Type\tIGORecommendation\tInvestigator Decision\n
+{report_str}
+""".format(
+                cmo_sample_name=cmo_sample_name, sample_id=sample_id, report_str=report_str
+            )
+
+            file_name = "{cmo_sample_name}_igo_qc_report".format(cmo_sample_name=cmo_sample_name)
+            sample_errors_event = UploadAttachmentEvent(self.job_group_notifier_id, file_name, msg).to_dict()
+            send_notification.delay(sample_errors_event)
+
+    def get_log_directory(self):
```

Comment:
I still need to investigate what this function does and migrate it over from argos to alignment pair accordingly